### PR TITLE
mgr/cephadm: ceph orch add fails when ipv6 address is surrounded by square brackets.

### PR DESCRIPTION
**Pull request by asm0deuz**
When calling "ceph orch add host1 [ipv6], it fails as the ipv6 address surrounded by [] is not recognized.

Fixes: https://tracker.ceph.com/issues/61885
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2153448

### CODE REVIEW

**Review by adk3798 — APPROVED** [review]
Makes sense to me, assuming you've tested this and it worked

**Comment by adk3798** [comment]
> Makes sense to me, assuming you've tested this and it worked

note that we have no CI testing for IPv6 because it doesn't work in the sepia lab currently afaik.

**Comment by asm0deuz** [comment]
@adk3798 my initial change didn't work as expected but now it works.

File: src/pybind/mgr/orchestrator/module.py [hunk 1]
```diff
@@ -460,7 +461,7 @@ def _add_host(self,
         if labels and len(labels) == 1:
             labels = labels[0].split(',')
 
-        s = HostSpec(hostname=hostname, addr=addr, labels=labels, status=_status)
+        s = HostSpec(hostname=hostname, addr=unwrap_ipv6(addr), labels=labels, status=_status)
```
Comment (adk3798) [inline src/pybind/mgr/orchestrator/module.py:464]:
```
orchestrator/module.py:464: error: Argument 1 to "unwrap_ipv6" has incompatible type "Optional[str]"; expected "str"
```

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by asm0deuz** [comment]
@adk3798 Done but still one job failing unfortunately I don't have the details

**Review by adk3798 — APPROVED** [review]
Seems like it should be okay, as long as it's fixing the issue on your side.

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-10-05_01:34:16-orch:cephadm-wip-adk-testing-2023-10-04-1525-distro-default-smithi/

reruns of failed jobs: https://pulpito.ceph.com/adking-2023-10-05_12:25:45-orch:cephadm-wip-adk-testing-2023-10-04-1525-distro-default-smithi/

After reruns, 1 failure

- https://tracker.ceph.com/issues/62958, known failure


Nothing to block merging

**Pull request merged**